Adding support for delayScale with multi metrics (#172)
DataDog/watermarkpodautoscaler: README.md

* **Scaling Delay**
<a name="delay"></a>

> In order to avoid scaling from bursts you can use the following features: `downscaleDelayBelowWatermarkSeconds` and/or `upscaleDelayAboveWatermarkSeconds`. These options are specified as integers. The metric(s) have to remain above or under its/their respective watermark for the configured duration.

Reviewer: This was better phrased before, in terms of translatability :)
```suggestion
In order to avoid scaling from bursts you can use the following features: `downscaleDelayBelowWatermarkSeconds` and/or `upscaleDelayAboveWatermarkSeconds`. These options are specified as integers. Each metric has to remain above or under its respective watermark for the configured duration.
```

Author: With `Each` my understanding is that we use `AND` and need all the metrics to stay above/below for the duration whereas we only need the union of the metrics to be above/below.

{"level":"info","ts":1668481092517.446,"logger":"controllers.WatermarkPodAutoscaler","msg":"Will not scale: value has not been out of bounds for long enough","watermarkpodautoscaler":"datadog/example-watermarkpodautoscaler","wpa_name":"example-watermarkpodautoscaler","wpa_namespace":"datadog","time_left":3209}
```

**Note:** If you are using multiple metrics with this feature, the above/below condition is considered using the `OR` of the metrics.

> e.g. if you have a 60s upscaleDelay with 2 metrics (M1 and M2), M1 stays above it's high watermark for 40s `[t0; t40]` and the M2 one goes above it's high watermark for 30s while overlapping with M1 during it's last 10s: `[t30; t60]`, this will validate the upscaleDelay condition and allow for an upscaling event.

Reviewer: ```suggestion
For example, suppose you have a 60 second `upscaleDelay` with two metrics, M1 and M2. If M1 stays above its high watermark for 40 seconds `[t0; t40]`, and the M2 one goes above its high watermark for 30 seconds, overlapping with M1 during its last 10 seconds, `[t30; t60]`, this validates the `upscaleDelay` condition and allows for an upscaling event.
```

## Limitations

> - Only officially supports one metric per WPA: While the logic supports multiple metrics and will apply the greatest recommendation of all metrics, the status needs some refactoring to reflect this insight.

Reviewer: ```suggestion
- Only officially supports one metric per WPA. While the logic supports multiple metrics and applies the greatest recommendation of all metrics, the status needs some refactoring to reflect this insight.
```
